src/Headless/OpenGLContext/OpenGLContext.cpp

```diff
@@ -62,6 +62,34 @@ OpenGLContext::OpenGLContext( const std::array<int, 2>& size ) {
             context->resizeFrameBuffer( width, height );
         };
         glfwSetFramebufferSizeCallback( m_glfwContext, resizeCB );
+        auto keyCB = []( GLFWwindow* window, int key, int scancode, int action, int mods ) {
+            auto context = static_cast<OpenGLContext*>( glfwGetWindowUserPointer( window ) );
+            context->keyboardEventCalback( key, scancode, action, mods );
+        };
+        glfwSetKeyCallback( m_glfwContext, keyCB );
+
+        auto mouseCB = []( GLFWwindow* window, int button, int action, int mods ) {
+            auto context = static_cast<OpenGLContext*>( glfwGetWindowUserPointer( window ) );
+            float xscale, yscale;
+            glfwGetWindowContentScale( window, &xscale, &yscale );
+            double xpos, ypos;
+            glfwGetCursorPos( window, &xpos, &ypos );
+            // TODO : verify if this is correct
+            // see https://www.glfw.org/docs/latest/window_guide.html#window_scale
+            context->mouseEventCalback(
+                button, action, mods, int( xpos / xscale ), int( ypos / yscale ) );
+        };
+        glfwSetMouseButtonCallback( m_glfwContext, mouseCB );
+
+        auto scrollCB = []( GLFWwindow* window, double xoffset, double yoffset ) {
+            auto context = static_cast<OpenGLContext*>( glfwGetWindowUserPointer( window ) );
+            float xscale, yscale;
+            glfwGetWindowContentScale( window, &xscale, &yscale );
+            // TODO : verify if this is correct
```

Comment:
This comment is a bit frustrating ;)
I looked at the doc, but I don't understand what could be wrong. Could you be more explicit ?

Reply:
I also read the doc and think this is correct. But as this was not tested, and due to past experience of wrong HDPI management (also called 1/4 window bug :p ) I just put here a reminder. But, sure, we can remove it without regrets.